projects/ardin_mb/state_handler.cc

```diff
@@ -367,14 +486,111 @@ bool StateHandler::checkAntPosition()
+    const auto &routeMax = m_Route.getMaxBound();
+
+    // Create vector field geometry to cover route bounds
+    m_VectorField.createVertices(routeMin[0] - 20_cm, routeMax[0] + 20_cm, 20_cm,
+                                 routeMin[1] - 20_cm, routeMax[1] + 20_cm, 20_cm);
+}
+//----------------------------------------------------------------------------
+bool StateHandler::handleUI()
+{
+    // Draw panoramic view window
+    if(ImGui::Begin("Panoramic", nullptr, ImGuiWindowFlags_NoResize))
+    {
+        ImGui::Image((void*)m_RenderTargetPanoramic.getTexture(),
+                     ImVec2(m_RenderTargetPanoramic.getWidth(), m_RenderTargetPanoramic.getHeight()),
+                     ImVec2(0, 1), ImVec2(1, 0));
+
+        if(ImGui::Button("Train")) {
+            m_VisualNavigation.train(m_SnapshotProcessor.getFinalSnapshot());
+        }
+        ImGui::SameLine();
+        if(ImGui::Button("Test")) {
+            std::cout << "Difference: " << m_VisualNavigation.test(m_SnapshotProcessor.getFinalSnapshot()) << std::endl;
+        }
+        ImGui::SameLine();
+        if(ImGui::Button("RIDF")) {
+            m_StateMachine.transition(State::BuildingRIDF);
+        }
     }
-    else {
-        m_Pose.x() = 5.0_m;
-        m_Pose.y() = 5.0_m;
-        m_Pose.yaw() = 270.0_deg;
+    ImGui::End();
+
+    // Draw top-down view window
+    if(ImGui::Begin("Top-down", nullptr, ImGuiWindowFlags_NoResize))
+    {
+        ImGui::Image((void*)m_RenderTargetTopDown.getTexture(),
+                     ImVec2(m_RenderTargetTopDown.getWidth(), m_RenderTargetTopDown.getHeight()),
+                     ImVec2(0, 1), ImVec2(1, 0));
+    }
+    ImGui::End();
+
+    // Draw processed snapshot view window
+    if(ImGui::Begin("Processed snapshot", nullptr, ImGuiWindowFlags_NoResize))
+    {
+        ImGui::Image((void*)m_FinalSnapshotTexture.getTexture(),
+                     ImVec2(m_VisualNavigation.getUnwrapResolution().width * 4, m_VisualNavigation.getUnwrapResolution().height * 4));
+    }
+    ImGui::End();
+
+    // Draw processed snapshot view window
+    if(ImGui::Begin("RIDF", nullptr, ImGuiWindowFlags_NoResize))
+    {
+        ImGui::PlotLines("", m_RIDFNovelty.data(), m_RIDFNovelty.size(), 0, nullptr,
+                         FLT_MAX, FLT_MAX, ImVec2(SimParams::numScanSteps * 5, 50));
+    }
+    ImGui::End();
+
+    if(ImGui::BeginMainMenuBar()) {
+        if(ImGui::BeginMenu("File")) {
+            if(ImGui::BeginMenu("Open Route")) {
+                char routeFilename[32];
+                for(unsigned int r = 1; r <= 14; r++) {
+                    sprintf(routeFilename, "ant1_route%u.bin", r);
+                    if(ImGui::MenuItem(routeFilename)) {
+                        const char *bobRoboticsPath = std::getenv("BOB_ROBOTICS_PATH");
+                        assert(bobRoboticsPath != nullptr);
+                        std::string routePath = std::string(bobRoboticsPath) + "/resources/antworld/";
```

Comment:
I'd use the helper function in ``path.h`` for this, partly so that you can get the correct path even when ``BOB_ROBOTICS_PATH`` isn't set.